install/gcp/deployment_manager/istio-cluster.jinja

```diff
@@ -129,12 +138,68 @@ resources:
           helm template install/kubernetes/helm/istio --name istio --namespace istio-system $ISTIO_OPTIONS  > istio.yaml
 
           kubectl apply -f istio.yaml
-          kubectl label namespace default istio-injection=enabled
 
-          sleep 150
+          {% if  properties['enableAutomaticSidecarInjection'] %}
+            kubectl label namespace default istio-injection=enabled
+          {% endif %}
+
+          kubectl rollout status deployment.extensions/istio-pilot -n istio-system --timeout=60s
+          kubectl rollout status deployment.extensions/istio-citadel -n istio-system --timeout=60s
+          kubectl rollout status deployment.extensions/istio-policy -n istio-system --timeout=60s
+          kubectl rollout status deployment.extensions/istio-galley -n istio-system --timeout=60s
+          kubectl rollout status deployment.extensions/istio-ingressgateway -n istio-system --timeout=60s
+
+          {% if  properties['enableKiali'] %}
+
+            export USERNAME=$(echo -n '{{  properties['kialiUsername'] }}' | base64)
+            export PASSPHRASE=$(echo -n '{{  properties['kialiPassword'] }}' | base64)
+            export NAMESPACE=istio-system
+
+            echo '
+            apiVersion: v1
+            kind: Secret
+            metadata:
+              name: kiali
+              namespace: $NAMESPACE
+              labels:
+                app: kiali
+            type: Opaque
+            data:
+              username: $USERNAME
+              passphrase: $PASSPHRASE
+            ' | envsubst > kaili_secret.yaml
+
+            kubectl apply -f kaili_secret.yaml
 
-          kubectl apply -f samples/bookinfo/platform/kube/bookinfo.yaml
-          kubectl apply -f samples/bookinfo/networking/bookinfo-gateway.yaml
+            export KIALI_OPTIONS=" --set kiali.enabled=true "
+
+            {% if  properties['enableGrafana'] %}
+              KIALI_OPTIONS=$KIALI_OPTIONS"  --set kiali.dashboard.grafanaURL=http://$(kubectl get svc grafana -n istio-system -o jsonpath='{.spec.clusterIP}'):3000"
+            {% endif %}
+
+            {% if  properties['enableTracing'] %}
+              KIALI_OPTIONS=$KIALI_OPTIONS" --set kiali.dashboard.jaegerURL=http://$(kubectl get svc tracing -n istio-system -o jsonpath='{.spec.clusterIP}'):80"
```

Comment:
I don't know if this is still valid.  Some of the addons have been changed to use port 443.  Can you manually verify this functions?

Reply:
you're right;  the `tracing.service.internalPort` is `:9411`
 
- https://istio.io/docs/reference/config/installation-options/

I'm not sure how to verify if Kiali->jaeger works.   The UI for kiali has the `distributed tracing` link in its UI but it points you to inte `ClusterIP` value within k8s and even then, towards the port for `zipkin`.   (eg the ui link from kiali tries to send you to `http://10.59.255.3:9411/`)

Reply:
@salrashid123 the istio.io documentation only represents istio 1.0.  preliminary.istio.io has more accurate documention, however, it is as of yet incomplete.  The cononical source is the helm charts themselves:
https://github.com/istio/istio/blob/release-1.1/install/kubernetes/helm/subcharts/gateways/values.yaml#L56-L69

Unfortunately those are the ingressgateway ports (which are enabled via HTTPS).  I would think that kiali->Jaeger should use the internal port address which is 9411, but as you point out is a ClusterIP.

@jmazzitelli can you provide feedback on this point?

Cheers
-steve

Reply:
basically, the jaeger URL and grafana URL just need to point to the exposed endpoints for those add-ons.

For reference, here's a couple places to see the grafana and jaeger integration - when the URLs are correct, you will see this (note: I'm looking at bookinfo "details" information in case you were wondering):

![addons ogv](https://user-images.githubusercontent.com/2029470/50781165-ac6e6b00-1272-11e9-897f-c85960c93eb0.gif)

Reply:
ah...ok, Wer'e not exposing any endpoints to the user in this script (all access is via tunnels) so the `jaeger` urls there through  `kiali` UI wont' work.  I can remove the settings in the DM that speify `kiali.dashboard.jaegerURL` and see if kiali just doesn't bother to show that tracing link or not.  For this DM script, we'll direct users to either view stats/traces in `stackdriver` or via tunnel to `grafana` and `jaeger`

i did verify metrircs shows up but thats just kiali collecting prometheus stats

![metrics](https://user-images.githubusercontent.com/11149054/50783798-b0918d00-1260-11e9-8ffb-b9d11e5e1a1b.png)

Reply:
or...since we already ask users to setup `port-foward` to `grafana` and `jaeger` already:

- Grafana
```
kubectl -n istio-system port-forward $(kubectl -n istio-system get pod -l app=grafana -o jsonpath='{.items[0].metadata.name}') 3000:3000
```

- Jaeger
```
kubectl -n istio-system port-forward $(kubectl -n istio-system get pod -l app=kiali -o jsonpath='{.items[0].metadata.name}') 20001:20001
```


how about we just set the exposed dashboard URLs to `localhost`?

```
{% if  properties['enableGrafana'] %}
   KIALI_OPTIONS=$KIALI_OPTIONS"  --set kiali.dashboard.grafanaURL=http://localhost:3000"
{% endif %}

{% if  properties['enableTracing'] %}
    KIALI_OPTIONS=$KIALI_OPTIONS" --set kiali.dashboard.jaegerURL=http://localhost:20001"
{% endif %}
```

Reply:
From Kiali's POV, we don't care what the URLs are, as long as they are accessible to the client browser. However you get that URL, Kiali is fine with it.

Of course, if you use "localhost" then the only Kiali user that can see the Jaeger/Grafana pages is the user whose browser is on the same box as the cluster.

Reply:
I'm fine w/ that (this is really only a quickstart script and we already ask users to proxy in).

Reply:
i set it to the internal `:16686`  address as described in:

- https://preliminary.istio.io/docs/reference/config/installation-options/#tracing-options